pkg/chain/ethereum/connect.go

```diff
@@ -60,7 +59,7 @@ type EthereumChain struct {
 
 // Connect performs initialization for communication with Ethereum blockchain
 // based on provided config.
-func Connect(accountKey *keystore.Key, config *ethereum.Config) (eth.Handle, error) {
+func Connect(accountKey *keystore.Key, config *ethereum.Config) (*EthereumChain, error) {
```

Comment:
Do we need to change it to `EhtereumChain` now?

Reply:
First, we need `EthereumChain` in `start.go` when we wrap it with tbtc extensions. Also, there are no benefits to return an interface here, I'd even say it doesn't fit here. The caller invokes `ethereum.Connect` to obtain a concrete ethereum chain instance, not a generic chain type. So that change is also in accordance with the Go `accept interfaces return structs` principle.